probe/appclient/app_client.go

```diff
@@ -311,13 +311,25 @@ func (c *appClient) startPublishing() {
 }
 
 // Publish implements Publisher
-func (c *appClient) Publish(r io.Reader) error {
+func (c *appClient) Publish(r io.Reader, shortcut bool) error {
 	// Lazily start the background publishing loop.
 	c.publishLoop.Do(c.startPublishing)
+	// enqueue report
 	select {
 	case c.readers <- r:
 	default:
 		log.Errorf("Dropping report to %s", c.hostname)
+		if shortcut {
+			return nil
+		}
+		// drop an old report to make way for new one
+		c.mtx.Lock()
+		defer c.mtx.Unlock()
+		select {
+		case <-c.readers:
+		default:
+		}
+		c.readers <- r
```

Comment:
Only for shortcut reports. For full reports we'd actually would rather have a single-place buffers.

I originally planned to do something along those lines, but it's a bigger change.